pages/search.tsx

```diff
@@ -1,24 +1,41 @@
-import { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import coingecko from '../utils/coingecko_helper';
+import { Coin } from '../types/Coin';
 
 const Search = () => {
-    const [newCoin, setNewCoin] = useState('')
-    const [coins, setCoins] = useState([])
-
-    const handleNewCoin = async (event: any) => {
-
+    const [newCoin, setNewCoin] = useState<string>('')
+    const [coins, setCoins] = useState<Coin[]>()
```

Comment:
```suggestion
    const [coins, setCoins] = useState<Coin[]>([])
```

Should generally initialise as an empty array.